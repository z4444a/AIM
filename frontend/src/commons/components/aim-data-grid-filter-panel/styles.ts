import { Theme } from '@material-ui/core/styles/createMuiTheme';
import createStyles from '@material-ui/core/styles/createStyles';
import { WithStyles } from '@material-ui/core/styles/withStyles';

export const styles = (theme: Theme) =>
  createStyles({
    container: {
      height: '100%',
      display: 'flex',
      flexDirection: 'column',
    },
  });

export type Style = WithStyles<'container'>;
