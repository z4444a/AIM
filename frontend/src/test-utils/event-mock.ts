export const getMockChangeEvent = (value: string) => {
  return {
    target: {
      value: value,
    },
  };
};

export const getMockKeyPressEvent = (key: string) => {
  return {
    key: key,
    preventDefault: jest.fn().mockName('preventDefault'),
  };
};
