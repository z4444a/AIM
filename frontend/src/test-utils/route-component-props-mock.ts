import { RouteComponentProps } from 'react-router';
import { Href } from 'history';

export function getMockRouterProps<P>(data: P): RouteComponentProps {
  const location = {
    hash: '',
    key: '',
    pathname: '',
    search: '',
    state: {},
  };

  const props: RouteComponentProps<P> = {
    match: {
      isExact: true,
      params: data,
      path: '',
      url: '',
    },
    location: location,
    history: {
      length: 2,
      action: 'POP',
      location: location,
      push: () => {},
      replace: () => {},
      go: () => {},
      goBack: () => {},
      goForward: () => {},
      block: () => () => {},
      createHref: () => {
        const temp: Href = '';
        return temp;
      },
      listen: () => () => {},
    },
    staticContext: {},
  };

  return props;
}
