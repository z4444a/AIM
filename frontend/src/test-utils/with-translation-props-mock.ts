import { WithTranslation } from 'react-i18next';

export function getMockWithTranslationProps(): WithTranslation {
  // eslint-disable-next-line
  const t = ((key: string) => `${key}_i18n`) as any;
  // eslint-disable-next-line
  const i18n = {} as any;
  const props: WithTranslation = {
    tReady: true,
    t,
    i18n,
  };

  return props;
}
