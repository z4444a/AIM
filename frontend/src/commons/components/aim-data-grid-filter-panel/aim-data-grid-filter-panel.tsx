import React, { ReactNode } from 'react';
import DialogTitle from '@material-ui/core/DialogTitle';
import DialogContent from '@material-ui/core/DialogContent';
import DialogActions from '@material-ui/core/DialogActions';
import Button from '@material-ui/core/Button';
import { WithTranslation } from 'react-i18next';
import { CustomFilterCmpContext, FilterValueType } from '../aim-data-grid/aim-data-grid';
import { Style } from './styles';

export interface State {
  filterValues: { [key: string]: FilterValueType };
}

export interface Props {
  open?: boolean;
  onClearFilters?: () => void;
  onApplyFilters?: (filterValues: { [key: string]: FilterValueType }) => void;
  renderCustomFilterCmp?: (context: CustomFilterCmpContext) => ReactNode;
  filterValues?: { [key: string]: FilterValueType };
}

export type InternalProps = Props & WithTranslation & Style;

export class AimDataGridFilterPanel extends React.PureComponent<InternalProps, State> {
  private readonly customFilterCmpContext: CustomFilterCmpContext;

  public constructor(props: InternalProps) {
    super(props);
    this.customFilterCmpContext = this.initCustomFilterCmpContext();
    this.state = { filterValues: {} };
  }

  public componentWillReceiveProps(nextProps: InternalProps) {
    this.setState({ filterValues: Object.assign({}, nextProps.filterValues) });
  }

  public render(): ReactNode {
    const { open, renderCustomFilterCmp, t, classes } = this.props;
    const { filterValues } = this.state;
    return (
      open && (
        <div className={classes.container}>
          <DialogTitle>{t('aimDataGrid.filterPanel.header')}</DialogTitle>
          <DialogContent>
            {typeof renderCustomFilterCmp === 'function' &&
              renderCustomFilterCmp({
                changeFilters: this.customFilterCmpContext.changeFilters,
                applyFilters: this.customFilterCmpContext.applyFilters,
                values: filterValues,
              })}
          </DialogContent>
          <DialogActions>
            <Button id="dataGridFilterPanelClearFiltersBtn" onClick={this.handleClearFilters}>
              {t('aimDataGrid.filterPanel.clearFiltersBtn')}
            </Button>
            <Button id="dataGridFilterPanelApplyFiltersBtn" onClick={this.handleApplyFilters}>
              {t('aimDataGrid.filterPanel.applyFiltersBtn')}
            </Button>
          </DialogActions>
        </div>
      )
    );
  }

  private initCustomFilterCmpContext(): CustomFilterCmpContext {
    const ctx: CustomFilterCmpContext = {
      changeFilters: values => {
        this.setState({ filterValues: values });
      },
      applyFilters: this.handleApplyFilters,
    };
    return ctx;
  }

  private handleClearFilters = () => {
    const { onClearFilters } = this.props;
    if (onClearFilters && typeof onClearFilters === 'function') {
      onClearFilters();
    }
  };

  private handleApplyFilters = () => {
    const { onApplyFilters } = this.props;
    const { filterValues } = this.state;
    if (onApplyFilters && typeof onApplyFilters === 'function') {
      onApplyFilters(filterValues);
    }
  };
}
