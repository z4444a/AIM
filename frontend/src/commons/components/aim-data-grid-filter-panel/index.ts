import { AimDataGridFilterPanel } from './aim-data-grid-filter-panel';
import { withTranslation } from 'react-i18next';
import withStyles from '@material-ui/core/styles/withStyles';
import { styles } from './styles';

export default withTranslation('common')(withStyles(styles)(AimDataGridFilterPanel));
