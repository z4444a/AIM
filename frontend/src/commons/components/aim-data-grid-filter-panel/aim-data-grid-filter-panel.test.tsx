import { mount, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import { AimDataGridFilterPanel, InternalProps } from './aim-data-grid-filter-panel';
import { getMockWithTranslationProps } from '../../../test-utils/with-translation-props-mock';
import sinon from 'sinon';

describe('AimDataGridFilterPanel', () => {
  function getComponentProps(open: boolean): InternalProps {
    const withTranslationProps = getMockWithTranslationProps();
    const props: InternalProps = {
      classes: {
        container: 'container',
      },
      ...withTranslationProps,
      open,
      renderCustomFilterCmp: () => <div id="innerComponent" />,
      onApplyFilters: sinon.spy(),
      onClearFilters: sinon.spy(),
    };
    return props;
  }

  function generateShallowWrapper(open: boolean) {
    return shallow(<AimDataGridFilterPanel {...getComponentProps(open)} />);
  }

  function generateMountWrapper(open: boolean) {
    return mount(<AimDataGridFilterPanel {...getComponentProps(open)} />);
  }

  it('should render correctly', () => {
    const wrapper = generateShallowWrapper(true);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should render internal filter component correctly', () => {
    const wrapper = generateShallowWrapper(true);
    expect(wrapper.find('#innerComponent').exists()).toBe(true);
  });

  it('should not render internal filter component', () => {
    const wrapper = generateShallowWrapper(false);
    expect(wrapper.find('#innerComponent').exists()).toBe(false);
  });

  it('should call clear filter fn', () => {
    const wrapper = generateMountWrapper(true);
    const clearFilterBtn = wrapper.find('#dataGridFilterPanelClearFiltersBtn').first();
    clearFilterBtn.simulate('click');
    expect(wrapper.props().onClearFilters.calledOnce).toBe(true);
  });

  it('should call apply filter fn', () => {
    const wrapper = generateMountWrapper(true);
    const clearFilterBtn = wrapper.find('#dataGridFilterPanelApplyFiltersBtn').first();
    clearFilterBtn.simulate('click');
    expect(wrapper.props().onApplyFilters.calledOnce).toBe(true);
  });
});
