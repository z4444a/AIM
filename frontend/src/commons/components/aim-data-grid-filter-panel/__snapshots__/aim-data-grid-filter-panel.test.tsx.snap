// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AimDataGridFilterPanel should render correctly 1`] = `
<div
  className="container"
>
  <WithStyles(DialogTitle)>
    aimDataGrid.filterPanel.header_i18n
  </WithStyles(DialogTitle)>
  <WithStyles(DialogContent)>
    <div
      id="innerComponent"
    />
  </WithStyles(DialogContent)>
  <WithStyles(DialogActions)>
    <WithStyles(Button)
      id="dataGridFilterPanelClearFiltersBtn"
      onClick={[Function]}
    >
      aimDataGrid.filterPanel.clearFiltersBtn_i18n
    </WithStyles(Button)>
    <WithStyles(Button)
      id="dataGridFilterPanelApplyFiltersBtn"
      onClick={[Function]}
    >
      aimDataGrid.filterPanel.applyFiltersBtn_i18n
    </WithStyles(Button)>
  </WithStyles(DialogActions)>
</div>
`;
